components/event-signup.js

```diff
@@ -0,0 +1,213 @@
+              </div>
+            )}
+          </div>
+          <div className="flex flex-col items-center">
+            {errors.attendance && (
+              <div className="text-red-600">
+                <small>{errors.attendance.message}</small>
+              </div>
+            )}
+            <div className="mt-1">
+              <input
+                type="radio"
+                id="inPerson"
+                name="attendance"
+                value={eventData?.inPerson}
+                {...register("attendance", {
+                  required:
+                    "Please let us know if you will be attending online or in person",
+                })}
+              />
+              <label htmlFor="inPerson">Attending In Person</label>
+            </div>
+            <div className="mb-2 mt-1">
+              <input
+                type="radio"
+                id="online"
+                name="attendance"
+                value={eventData?.virtual}
+                {...register("attendance", {
+                  required:
+                    "Please let us know if you will be attending online or in person",
+                })}
+              />
+              <label htmlFor="online">Joining Online</label>
+            </div>
+          </div>
+          <button
+            type="submit"
+            className="m-auto w-full rounded-md bg-gray-900 px-7 py-4 font-semibold text-white transition-colors hover:bg-gray-800 focus:outline-none focus:ring focus:ring-gray-200 focus:ring-offset-2 dark:bg-white dark:text-black md:w-[25%]"
+          >
+            {isSubmitting ? (
+              <svg
+                className="mx-auto h-5 w-5 animate-spin text-white dark:text-black"
+                xmlns="http://www.w3.org/2000/svg"
+                fill="none"
+                viewBox="0 0 24 24"
+              >
+                <circle
+                  className="opacity-25"
+                  cx="12"
+                  cy="12"
+                  r="10"
+                  stroke="currentColor"
+                  strokeWidth="4"
+                ></circle>
+                <path
+                  className="opacity-75"
+                  fill="currentColor"
+                  d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
+                ></path>
+              </svg>
+            ) : (
+              "Register"
+            )}
+          </button>
+        </form>
+        {isSubmitSuccessful && registerSuccess && (
+          <div className="mt-3 text-center">
+            {"Registration Success. Here's your ticket:"}
+            <a className="block text-blue-600" target="_blank" href={ticketUrl}>
+              {ticketUrl}
+            </a>
+          </div>
+        )}
+        {isSubmitSuccessful && registerSuccess === false && (
+          <div className="mt-3 text-center text-sm text-red-500">
+            {"Something went wrong. Please try later."}
```

Comment:
Use an error state to render an error. There are two different api calls here and either can fail.

Reply:
Resolved in commit: c18ad30b5eefc35725dbdffc4510cd29641bddd4